pandas/core/arrays/datetimelike.py

```diff
@@ -437,25 +690,122 @@ def _values_for_factorize(self):
     def _from_factorized(cls, values, original):
         return cls(values, dtype=original.dtype)
 
+    def _values_for_argsort(self):
+        return self._data
+
     # ------------------------------------------------------------------
-    # Null Handling
+    # Additional array methods
+    # These are not part of the EA API, but we implement them because
+    # pandas currently assumes they're there.
 
-    def isna(self):
-        return self._isnan
+    def view(self, dtype=None):
+        """
+        New view on this array with the same data.
 
-    @property  # NB: override with cache_readonly in immutable subclasses
-    def _isnan(self):
+        Parameters
+        ----------
+        dtype : numpy dtype, optional
+
+        Returns
+        -------
+        ndarray
+            With the specified `dtype`.
         """
-        return if each value is nan
+        return self._data.view(dtype=dtype)
+
+    def value_counts(self, dropna=False):
         """
```

Comment:
isn't this on EA base? I think we had this discussion, why isn't it?

Reply:
It's not. We were discussing removing it, since it doesn't really feel like an array method. https://github.com/pandas-dev/pandas/issues/22843.

It's replacement would be a way to ensure that the correct values are passed to `pandas.value_counts`, and the the index is constructed properly.